### PR TITLE
fix mul, generator and add hardcoded tests

### PR DESCRIPTION
**Pull request by lonerapier**
fixes mul and multiplicative generator of extension field

### CODE REVIEW

File: src/field/gf_101_2.rs [hunk 1]
```diff
@@ -52,24 +50,26 @@ impl<F: FiniteField> From<F> for QuadraticPlutoField<F> {
 impl<F: FiniteField> FiniteField for QuadraticPlutoField<F> {
   type Storage = u32;
 
+  const NEG_ONE: Self = Self::new(F::NEG_ONE, F::ZERO);
+  // TODO: This is wrong
+  const ONE: Self = Self::new(F::ONE, F::ZERO);
```
Comment (0xJepsen) [inline src/field/gf_101_2.rs:55]: What should this be?

File: src/field/gf_101_2.rs [hunk 2]
```diff
@@ -322,7 +340,7 @@ mod tests {
 
   #[test]
   fn test_generator_order() {
-    let generator = F2::generator();
+    let generator = F2::ONE;
     let mut x = generator;
     for _ in 1..F2::ORDER {
       x *= generator;
```
Comment (0xJepsen) [inline src/field/gf_101_2.rs:346]: This isn't correct right? We are just multiplying one together a bunch of times, this test would pass no matter what right?

**Review by 0xJepsen — APPROVED** [review]

File: src/field/gf_101_2.rs (hunk 2 above)
Comment (lonerapier) [reply src/field/gf_101_2.rs]: oops lol yes, how did i miss this. fixed it now. Now, we're multiplying 1 by generator, order-1 times, which is the multiplicative order of the group defined by field.

File: src/field/gf_101_2.rs (hunk 1 above)
Comment (lonerapier) [reply src/field/gf_101_2.rs:55]: i think it's a clippy issue, it reorganised the comments somehow, not sure.

**Pull request merged**